presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -601,4 +601,18 @@ public static Decimal toMetastoreDecimal(BigDecimal decimal)
     {
         return new Decimal(ByteBuffer.wrap(decimal.unscaledValue().toByteArray()), (short) decimal.scale());
     }
+
+    /**
+     * Metastore stores NDV considering null as a distinct value
+     */
+    private static OptionalLong toMetastoreDistinctValuesCount(OptionalLong distinctValuesCount, OptionalLong nullsCount)
+    {
+        if (distinctValuesCount.isPresent() && nullsCount.isPresent()) {
+            if (nullsCount.getAsLong() > 0) {
+                return OptionalLong.of(distinctValuesCount.getAsLong() + 1);
+            }
+            return distinctValuesCount;
+        }
+        return OptionalLong.empty();
```

Comment:
This seems wrong. If `distinctValuesCount` is present but `nullsCount` is missing, we return empty. If that is intentional, document why.

I would expect the algorithm to be
```java
if (distinctValuesCount.isPresent() && nullsCount.isPresent() && (nullsCount.getAsLong() > 0)) {
    return OptionalLong.of(distinctValuesCount.getAsLong() + 1);
}
return distinctValuesCount;
```